static/app/views/releases/thresholdsList/thresholdGroupRows.tsx

```diff
@@ -213,56 +206,45 @@ export function ThresholdGroupRows({
   return (
     <StyledThresholdGroup>
       {Array.from(thresholdIdSet).map((tId: string, idx: number) => {
-        const threshold = editingThresholds[tId] || thresholdsById[tId];
-        let environmentName;
-        if (editingThresholds[tId]) {
-          // editing environment type is String
-          environmentName = editingThresholds[tId].environment;
+        let threshold;
+        let isEditing = false;
```

Comment:
nit: you can keep these as const with inline conditions like:
```
const isEditing = tId in editingThresholds;
const threshold = tId in editingThresholds ? editingThresholds[tId] as EditingThreshold : initialThreshold as Threshold
```

Reply:
yea mmm.
This is me kinda cheating.
this is casting threshold as `any` to avoid the switch ternary below 🤔 
I'll redo this

Reply:
Ok - so instead i've explicitly cast the types below :( 
not great, but unfortunatley TS compiler isn't smart enough to dynamically derive the state